### PR TITLE
ARM: dts: bcm2835: Switch HSM clock to firmware

### PR DESCRIPTION
**Pull request by mripard**
This is a PR to fix a stall on the Pi3 when it boots without an HDMI display connected.

It was reported in the PR #4940

### CODE REVIEW

**Comment by pelwell** [comment]
The first sentence of the commit message is hard to follow, but the patch looks OK.

**Comment by popcornmix** [comment]
I can confirm this fixes the hang on boot on Pi3+ with hdmi unplugged.

**Comment by pelwell** [comment]
Merged with a slightly tweaked commit message.
